studybuddies/src/app/components/signup/signup.component.ts

```diff
@@ -0,0 +1,38 @@
+import { Component, OnInit } from '@angular/core';
+import { AuthService } from '../services/auth.service'
+import { Router } from '@angular/router';
+
+@Component({
+  selector: 'app-signup',
+  templateUrl: './signup.component.html',
+  styleUrls: ['./signup.component.css']
+})
+export class SignupComponent implements OnInit {
+
+  user = {
+    name: '',
+    email: '',
+    password: ''
+  }
+
+  constructor(
+    private authService: AuthService,
+    private router: Router
+    ) { }
+
+  ngOnInit(): void {
+  }
+
+  signUp(){
+    this.authService.signUp(this.user)
+      .subscribe(
+        res => {
+          console.log(res);
+          localStorage.setItem('token', res.token);
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Team-Studybuddies/Studybuddies-Frontend/pullRequest?prid=7163724)
```suggestion
          localStorage.setItem("token", res.token);
```